src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+				if (!response.isEmpty()) {
+					objectMapper = new ObjectMapper();
+					DeviceInfo deviceInfo = objectMapper.readValue(response.toString(), DeviceInfo.class);
+
+					String deviceInfoGroup = MonitoringMetricGroup.DEVICE_INFO.getName() + DecoderConstant.SPACE + DecoderConstant.HASH;
+
+					stats.put(deviceInfoGroup + DeviceInfoMetric.SERIAL_NUMBER.getName(), checkForNullData(deviceInfo.getSerialNumber()));
+					stats.put(deviceInfoGroup + DeviceInfoMetric.HARDWARE_REVISION.getName(), checkForNullData(deviceInfo.getHardwareRevision()));
+					stats.put(deviceInfoGroup + DeviceInfoMetric.PART_NUMBER.getName(), checkForNullData(deviceInfo.getPartNumber()));
+					stats.put(deviceInfoGroup + DeviceInfoMetric.FIRMWARE_VERSION.getName(), checkForNullData(deviceInfo.getFirmwareVersion()));
+					stats.put(deviceInfoGroup + DeviceInfoMetric.TEMPERATURE.getName(), checkForNullData(deviceInfo.getTemperature()));
+				} else {
+					updateDeviceInfoFailedMonitor(failedMonitor);
+				}
+			} else {
+				updateDeviceInfoFailedMonitor(failedMonitor);
+			}
+		} catch (Exception e) {
+			updateDeviceInfoFailedMonitor(failedMonitor);
+		}
+	}
+
+	/**
+	 * Update failedMonitor with Getting decoder stats error message
+	 *
+	 * @param failedMonitor list statistics property
+	 * @param decoderID ID of the decoder
+	 */
+	private void updateDecoderStatisticsFailedMonitor(Map<String, String> failedMonitor, Integer decoderID) {
+		failedMonitor.put(MonitoringMetricGroup.DECODER_STATISTICS.getName() + decoderID, DecoderConstant.GETTING_DECODER_STATS_ERR + decoderID);
+	}
+
+	/**
+	 * This method is used to retrieve decoder statistic by send GET request to http://{IP_Address}/apis/decoders/:id
+	 *
+	 * @param stats list statistics property
+	 * @param decoderID ID of the decoder
+	 *
+	 * When sessionID is null, the failedMonitor is going to update
+	 * When there is no response data, the failedMonitor is going to update
+	 * When there is an exception, the failedMonitor is going to update and exception is not populated
+	 */
+	private void retrieveDecoderStats(Map<String, String> stats, Integer decoderID) {
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				JsonNode response = doGet(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.DECODERS + DecoderConstant.SLASH + decoderID), JsonNode.class);
+				if (!response.isEmpty()) {
+					objectMapper = new ObjectMapper();
+					DecoderData decoderData = objectMapper.readValue(response.toString(), DecoderData.class);
+					DecoderStats decoderStats = decoderData.getDecoderStats();
+					DecoderInfo decoderInfo = decoderData.getDecoderInfo();
+					List<AudioPair> audioPairs = decoderStats.getAudioPairs();
+
+					String decoderStatisticGroup = MonitoringMetricGroup.DECODER_STATISTICS.getName() + DecoderConstant.SPACE + decoderID + DecoderConstant.HASH;
+
+					// Update static stats
```

Comment:
Please add it into a loop. Something like:
`for(DecoderMonitoringMetric item : DecoderMonitoringMetric.values()){
  stats.put(decoderStatisticGroup + item.getName(), 
                    checkForNullData(DecoderStats.getValueByMonitoringMetric(item)));
}`

Reply:
I fixed it